src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,119 @@
+import { FC, useState } from "react";
+import {
+  ActionButton,
+  Button,
+  Icon,
+  Input,
+  Modal,
+  Notification,
+} from "@canonical/react-components";
+import { useFormik } from "formik";
+import { createFineGrainedTlsIdentity } from "api/auth-identities";
+import { base64EncodeTokenDetails } from "util/helpers";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [token, setToken] = useState<string | null>(null);
+  const [error, setError] = useState(false);
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+
+  const formik = useFormik({
+    initialValues: {
+      clientName: "",
+    },
+    onSubmit: (values) => {
+      setIsLoading(true);
+
+      createFineGrainedTlsIdentity(values.clientName)
+        .then((tokenPreResponse) => {
+          const tempToken = base64EncodeTokenDetails(tokenPreResponse);
+
+          setIsLoading(false);
+          setToken(tempToken);
+
+          void queryClient.invalidateQueries({
+            queryKey: [queryKeys.identities],
+          });
+        })
+        .catch(() => {
+          setIsLoading(false);
+          setError(true);
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={onClose}
+      className="create-tls-identity"
+      title="Generate trust token"
+      buttonRow={
+        <>
+          {token && (
+            <Button
+              className="u-no-margin--bottom"
+              onClick={() => {
+                navigator.clipboard.writeText(token);
+              }}
+              type="button"
+              hasIcon
+            >
+              <Icon name="copy" />
```

Comment:
This should have a copy label as well, to be discoverable to screen readers.

Reply:
This is still not resolved.

Please do not resolve comments yourself. Better add comment with a reference to the commit that fixes the issue that was pointed out.

Reply:
Hi David,
You asked for a label to be added for screen readers, and I added an aria-label. Was there anything else you wanted to add?
I resolve comments as I make code changes to keep track of my progress and minimise clutter on the screen which can be confusing.